src/sentry/workflow_engine/models/detector.py

```diff
@@ -105,3 +106,28 @@ def detector_handler(self) -> DetectorHandler | None:
     def get_audit_log_data(self) -> dict[str, Any]:
         # TODO: Create proper audit log data for the detector, group and conditions
         return {}
+
+    def get_option(
+        self, key: str, default: Any | None = None, validate: Callable[[object], bool] | None = None
+    ) -> Any:
+        if not self.project:
+            raise ValueError("Detector must have a project to get options")
+
+        return self.project.get_option(key, default=default, validate=validate)
+
+
+class ErrorDetector(Detector):
```

Comment:
nit: i think we'd want to move this class out of this file, my guess is we're eventually going to have a lot of detectors.